### PR TITLE
Simplifying class ExpectFileGenerator to function - master

### PR DESCRIPTION
**Pull request by Bischoff**

### CODE REVIEW

File: features/support/commonlib.rb [hunk 1]
```diff
@@ -3,34 +3,25 @@
...
-    File.basename(path)
+# generate expect file to interactively run spacewalk-push-register
+def generate_expect_file(host, bootstrap)
+  raise(ArgumentError, 'Hostname is missing!') if host.nil? || host.empty?
+  path = ''
+  Tempfile.open('push-registration.expect') do |f|
+    path = f.path
+    f.write('spawn spacewalk-ssh-push-init --client ' + host + ' --register ' + bootstrap + ' --tunnel' + "\n")
+    f.write("while {1} {\n")
+    f.write("  expect {\n")
```
Comment (MalloZup) [inline features/support/commonlib.rb]: Create a constant Var like TEXT = "blabla".freeze, note the freeze in ruby, so we remove data that changes from code logic and we have only 1 time file writing

File: features/support/commonlib.rb [hunk 2]
```diff
@@ -3,34 +3,25 @@
 
 require 'tempfile'
 
-# Class for generating expect files to interactively run spacewalk-push-register.
-class ExpectFileGenerator
-  def initialize(host, bootstrap)
-    raise(ArgumentError, 'Hostname is missing!') if host.nil? || host.empty?
-    Tempfile.open('push-registration.expect') do |f|
-      @file = f
```
Comment (MalloZup) [inline features/support/commonlib.rb]:
```ruby
BOOTSTRAP_GINO = " "while {1} {\n   expect {\n" \
                            \"Are you sure you want to continue connecting (yes/no)?\" {send \"yes\r\"} \
                             "BLABLA".freeze                           

f.write(BOOTSTRAP_GINO)
```

File: features/support/commonlib.rb [hunk 3]
```diff
@@ -3,34 +3,25 @@
...
-  # Return the file name.
-  def filename
-    File.basename(path)
+# generate expect file to interactively run spacewalk-push-register
+def generate_expect_file(host, bootstrap)
+  raise(ArgumentError, 'Hostname is missing!') if host.nil? || host.empty?
+  path = ''
+  Tempfile.open('push-registration.expect') do |f|
+    path = f.path
+    f.write('spawn spacewalk-ssh-push-init --client ' + host + ' --register ' + bootstrap + ' --tunnel' + "\n")
```
Comment (MalloZup) [inline features/support/commonlib.rb]:
```ruby
BOOTSTRAP_GINO =  "while {1} {\n   expect {\n" \
                            \"Are you sure you want to continue connecting (yes/no)?\" {send \"yes\r\"} \
                             "BLABLA".freeze                           

f.write(BOOTSTRAP_GINO)
```

File: features/support/commonlib.rb [hunk 4]
```diff
@@ -3,34 +3,23 @@
...
-  end
-end
-
+# return current URL
 def current_url
   driver.current_url
 end
+
+# generate expect file to interactively run spacewalk-push-register
+def generate_expect_file(host, bootstrap)
```
Comment (MalloZup) [inline features/support/commonlib.rb]: we can make this more generic with injecting the text as an argument
```ruby
def generate_expect_file(host, bootstrap, script
```
and we have the script outside the function, so you can use this function even for the proxy creation

**Comment by Bischoff** [comment]
Good idea.

**Pull request merged**